src/com/amazon/ion/impl/_Private_LSTWriter.java

```diff
@@ -0,0 +1,300 @@
+package com.amazon.ion.impl;
+
+import com.amazon.ion.IonCatalog;
+import com.amazon.ion.IonReader;
+import com.amazon.ion.IonType;
+import com.amazon.ion.IonValue;
+import com.amazon.ion.SymbolTable;
+import com.amazon.ion.SymbolToken;
+import com.amazon.ion.Timestamp;
+
+import java.io.IOException;
+import java.math.BigDecimal;
+import java.math.BigInteger;
+import java.util.Date;
+import java.util.List;
+import java.util.LinkedList;
+
+@Deprecated
+public class _Private_LSTWriter implements _Private_IonWriter {
+    private int depth;
+    private SymbolTable symbolTable;
+    private State state;
+    private LinkedList<SSTImportDescriptor> declaredImports;
+    private LinkedList<String> declaredSymbols;
+    private final IonCatalog catalog;
+    private boolean seenImports;
+    private boolean seenSymbols;
+
+    /**
+     * Enum for managing our state between API calls.
+     * <p>
+     * LST_STRUCT: we are in an lst struct at depth 1.
+     * PRE_IMPORT_LIST: setfield was called with the symbol 'imports' from LST_STRUCT.
+     * IMPORT_LIST: either stepin(List) was called from PRE_IMPORT_LIST, or stepOut from IMPORT_DESCRIPTOR.
+     * IMPORT_DESCRIPTOR: StepIn(Struct) from IMPORT_LIST was called or a field/value pair was written.
+     * IMPORT_MAX_ID: setfield was called with the symbol 'max_id' from IMPORT_DESCRIPTOR.
+     * IMPORT_NAME: setfield was called with the symbol 'name' from IMPORT_DESCRIPTOR.
+     * IMPORT_VERSION: setfield was called with the symbol 'version' from IMPORT_DESCRIPTOR.
+     * PRE_SYMBOL: setfield was called with the symbol 'symbols' from LST_STRUCT.
+     * SYMBOL_LIST: stepin(List) was called from PRE_SYMBOL_LIST.
+     */
+    private enum State {LST_STRUCT, PRE_IMPORT_LIST, IMPORT_LIST, IMPORT_DESCRIPTOR, IMPORT_MAX_ID, IMPORT_NAME, IMPORT_VERSION, PRE_SYMBOL_LIST, SYMBOL_LIST}
+
+    /**
+     * Holds import descriptor info between API calls.
+     **/
+    private static class SSTImportDescriptor {
+        String name;
+        int version;
+        int maxID;
+    }
+
+    /**
+     * Constructor for the _Private_LSTWriter
+     * <p>
+     * Absorbs system processing from user level APIs in the IonManagedBinaryWriter.
+     *
+     * @param  imports ArrayList<SymbolTable> is populated by the builder.
+     * @param  symbols List<String>
+     * @param  inCatalog IonCatalog
+     */
+   public _Private_LSTWriter(List<SymbolTable> imports, List<String> symbols, IonCatalog inCatalog) {
+        catalog = inCatalog;
+        declaredSymbols = new LinkedList<String>();
+        if(imports.isEmpty() || !imports.get(0).isSystemTable()){
```

Comment:
We are absorbing system processing values that update our imported sst context. The managed binarywriter uses a different import list called fallbackimports for future use after finish is called.
The imports passed into the lst is a copy of the one provided by the builder it is not altering anything other machinery would touch.

Reply:
Your statement is true now, however it may break in the future. If you still think it's important to keep the behavior document it in the constructor javadoc.